From cfbb9e1d04acc715707188dd7ca7fdb2ede9d7b8 Mon Sep 17 00:00:00 2001
From: Michael Matz <matz@suse.de>
Date: Tue, 2 Apr 2013 04:39:56 +0200
Subject: [PATCH 138/169] Fix using uninitialized value

TCG can't cope with uninitialized temps used as inputs, even when
the whole sequence will overwrite the whole content.  It silentely
overwrites registers that still hold useful values.  This fixes
segfaults in python3-base testsuite.
---
 target-arm/translate-a64.c | 3 +++
 1 file changed, 3 insertions(+)

Index: qemu-1.6.0+dfsg/target-arm/translate-a64.c
===================================================================
--- qemu-1.6.0+dfsg.orig/target-arm/translate-a64.c	2013-11-05 22:23:44.179361589 +0000
+++ qemu-1.6.0+dfsg/target-arm/translate-a64.c	2013-11-05 22:23:44.175361599 +0000
@@ -3036,6 +3036,9 @@
 	    unallocated_encoding(s);
 	    return;
 	}
+	/* We need to initialize tcg_res, even though we overwrite
+	   all its bytes, otherwise TCG will generate wrong code.  */
+	tcg_gen_movi_i64(tcg_res, 0);
 	for (i = 0; i < 8; i += ebytes) {
 	    /* XXX Should we be able to just generate a smaller
 	       load?  */
